frontend/src/modules/member/components/view/member-view-contributions.vue

```diff
@@ -1,220 +1,613 @@
   let topicMap = {}
   props.contributions.forEach((contribution) => {
+    // Extract the name of the contribution from the URL
+    const name = contribution.url.split('/').pop()
     contribution.topics.forEach((topic) => {
       if (!topicMap[topic]) {
-        topicMap[topic] = [contribution.id]
+        topicMap[topic] = [name]
       } else {
-        topicMap[topic].push(contribution.id)
+        topicMap[topic].push(name)
       }
     })
   })
+
+  // Next, the topics in the topicMap are iterated over and for each topic,
+  // the corresponding contribution IDs are used to create edges
+  // between the contributions.
   for (let topic in topicMap) {
     let contributionIds = topicMap[topic]
+
     for (let i = 0; i < contributionIds.length; i++) {
       for (let j = i + 1; j < contributionIds.length; j++) {
-        if (contributionIds[i] !== contributionIds[j]) {
-          edges[topic] = {
+        const id = [contributionIds[i], contributionIds[j]]
+          .sort() // Sort the IDs so that the edge ID is consistent
+          .join('-')
+        if (contributionIds[i] === contributionIds[j]) {
+          break
+        }
+        if (!edges[id]) {
+          edges[id] = {
             source: contributionIds[i].toString(),
-            target: contributionIds[j].toString()
+            target: contributionIds[j].toString(),
+            topics: [topic],
+            size: 1
           }
+        } else {
+          if (edges[id].topics.indexOf(topic) === -1) {
+            edges[id].topics.push(topic)
+          }
+          edges[id].size = Math.min(
+            ((3 - 1) / (10 - 1)) *
+              (edges[id].topics.length - 1) +
+              1,
+            3
+          )
         }
       }
     }
   }
-  console.log(edges)
   return edges
 })
+
+// This code uses computed properties to calculate the center position
+// of a specific edge on a graph
+const edgeCenterPos = computed(() => {
+  // It first checks if the edge exists by checking if
+  // the value of the targetEdgeId variable is present in the edges array
+  // If it does not exist, it returns an object with x and y values of 0.
+  const edge = edges.value[targetEdgeId.value]
+  if (!edge) return { x: 0, y: 0 }
+
+  // If it does exist, it assigns the source and target nodes of the edge to variables.
+  const sourceNode = edges.value[targetEdgeId.value].source
+  const targetNode = edges.value[targetEdgeId.value].target
+
+  // Then, it calculates the x and y values of the center position
+  // by taking the average of the x and y values of the source and target nodes.
+  return {
+    x:
+      (layouts.value.nodes[sourceNode].x +
+        layouts.value.nodes[targetNode].x) /
+      2,
+    y:
+      (layouts.value.nodes[sourceNode].y +
+        layouts.value.nodes[targetNode].y) /
+      2
+  }
+})
+
+// If either we are on a node or an edge, the graph is disabled
+const isDisabled = computed(() => {
```

Comment:
Do we need to disable the graph when we are interacting with a node/edge?
I'm not sure if it's because of this but if I have a tooltip open and I click on another node, I will have to click it twice to open a new tooltip. First click to close the previous tooltip and another to open the new one